docs/walkthrough/walkthrough-oauth2-proxy.md

```diff
@@ -0,0 +1,168 @@
+
+To create the OAuth App on Github, browse https://github.com/settings/developers and click the `New OAuth App` button.
+
+Fill in the form and click the `Register application` button:
+- `Application name`: the name of the application
+- `Homepage URL`: Set this to `http://auth.127.0.0.1.nip.io` but this in not important
+- `Authorization callback URL`: Set this to `http://auth.127.0.0.1.nip.io/oauth2`, we will expose `oauth2-proxy` under this url in the next step
+
+![Register OAuth application](./walkthrough-oauth2-proxy-register.png)
+
+Once the apllication is registered, you will obtain a `Client ID` and `Client Secret`.
+
+This will be needed to configure `oauth2-proxy` in the next step.
+
+![Registered OAuth application](./walkthrough-oauth2-proxy-registered.png)
+
+## Installing and configuring oauth2-proxy
+
+Now you have the OAuth application created on Github, you can deploy [oauth2-proxy](https://github.com/oauth2-proxy/oauth2-proxy) in your cluster.
+
+You will install it using helm. The helm chart used in this guide is hosted on the [helm official reepository](https://github.com/helm/charts/tree/master/stable/oauth2-proxy).
+
+To install the chart, simply set the `Client ID` and `Client Secret` you just obtained from Github and run the command below:
+
+```bash
+# CLIENT_ID and CLIENT_SECRET are the Client ID and Client Secret obtained
+# when creating the Github OAuth application in the previous step
+CLIENT_ID=__THE_CLIENT_ID_OF_YOUR_GITHUB_OAUTH_APP__
+CLIENT_SECRET=__THE_CLIENT_SECRET_OF_YOUR_GITHUB_OAUTH_APP__
+
+cat <<EOF | helm upgrade --install --wait --create-namespace --namespace tools oauth2-proxy stable/oauth2-proxy --values -
+config:
+  clientID: $CLIENT_ID
+  clientSecret: $CLIENT_SECRET
+
+extraArgs:
+  provider: github
+  whitelist-domain: .127.0.0.1.nip.io
+  cookie-domain: .127.0.0.1.nip.io
+  redirect-url: http://auth.127.0.0.1.nip.io/oauth2/callback
+  cookie-secure: 'false'
+
+ingress:
+  enabled: true
+  path: /
+  hosts:
+    - auth.127.0.0.1.nip.io
+EOF
+```
+
+This will install `oauth2-proxy` in your cluster in the `tools` namespace.
+
+Some parameters in this configuration deserve a bit of explanations:
+- `config.ClientID` and `config.clientSecret` are obtained when creating the Github OAuth application
+- `extraArgs.provider` tells `oauth2-proxy` to use Github
+- `extraArgs.redirect-url` is needed for HTTP callbacks, `oauth2-proxy` will default to HTTPS if not set
+- `extraArgs.whitelist-domain` allows `oauth2-proxy` to authenticate request for all `.127.0.0.1.nip.io` subdomains
+- `extraArgs.cookie-secure` disables CSRF as we are using HTTP (not HTTPS)
+- `extraArgs.cookie-domain` is used to set the cookie for all `.127.0.0.1.nip.io` subdomains. As `oauth2-proxy` is deployed centrally, it will be exposed at `http://auth.127.0.0.1.nip.io` while the dashboard will be exposed at `http://tekton-dashboard.127.0.0.1.nip.io`. The cookie needs to be available to both urls
```

Comment:
```suggestion
- `extraArgs.cookie-domain` is used to set the cookie for all `.127.0.0.1.nip.io` subdomains. As `oauth2-proxy` is deployed centrally, it will be exposed at `http://auth.127.0.0.1.nip.io` while the dashboard will be exposed at `http://tekton-dashboard.127.0.0.1.nip.io`. The cookie needs to be available to both URLs
```